polygeist/tools/cgeist/driver.cc

```diff
@@ -923,16 +923,19 @@ static void processInputFiles(const cl::list<std::string> &inputFiles,
             triple, DL, commands);
 }
 
-static bool hasDeviceFuncs(mlir::gpu::GPUModuleOp deviceModule) {
-  return !deviceModule.getRegion().getOps<mlir::gpu::GPUFuncOp>().empty();
+static bool containsFunctions(mlir::gpu::GPUModuleOp deviceModule) {
+  Region &rgn = deviceModule.getRegion();
+  return !rgn.getOps<mlir::gpu::GPUFuncOp>().empty() ||
+         !rgn.getOps<mlir::func::FuncOp>().empty();
```

Comment:
I also thought of this change :+1:. As the device module will be empty if we do not add functions to it, maybe we can do something like:

```
return !deviceModule.getRegion().empty();
```

Reply:
I tried that actually and, surprisingly, the region is consider non-empty even if it contains nothing.

Reply:
That is weird. Fine!